core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ast/StatementMatcher.java

```diff
@@ -170,6 +170,24 @@ public int hashCode() {
 		return Objects.hash(subjectName, subjectValue, predicateName, predicateValue, objectName, objectValue);
 	}
 
+	public static class StableRandomVariableProvider {
+		private static final String BASE = UUID.randomUUID().toString().replace("-", "") + "_"; // We just need a random
+																								// base that isn't used
+																								// elsewhere in the
+																								// ShaclSail, but we
+																								// don't want it to be
+																								// stable so we can
+																								// compare the SPARQL
+																								// queries where these
+																								// variables are used
```

Comment:
Code formatter went a bit haywire here I reckon.

Reply:
Huh, weird. Least effort is probasbly to put this comment above instead of after the line. More reabable anyway.